maven-core/src/main/java/org/apache/maven/DefaultMaven.java

```diff
@@ -495,6 +495,7 @@ private void validatePrerequisitesForNonMavenPluginProjects(List<MavenProject> p
      * @return A {@link Set} of profile identifiers, never {@code null}.
      */
     private Set<String> getAllProfiles(MavenSession session) {
+        // TODO: which version of super pom should we use here ?
```

Comment:
Maybe the same version that project uses? But: what happens if reactor has multiple versions (not same)? -- IMHO should fail.

Reply:
I've update this part of the computation to be coherent with the way superPoms are used during model building.  The superPom of a given project is determined by the modelVersion of that project and not the modelVersion of its upper parent.  What this means is that if you write a 4.1.0 project which inherit a 4.0.0 parent, the superPom will be 4.1.0.